util/bufferPool.go

```diff
@@ -0,0 +1,23 @@
+package util
+
+import (
+	"sync"
+)
+
+type BufferPool struct {
```

Comment:
while it's good that types are named camelcase, I think we should keep filenames like buffer_pool.go that seems to be the convention